hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -681,20 +684,63 @@ void finalizeJob(@Nullable Throwable failure) {
     }
 
     private boolean isSuccess(@Nullable Throwable failure) {
-        long elapsed = NANOSECONDS.toMillis(System.nanoTime() - executionStartTime);
         if (failure == null) {
-            logger.info(String.format("Execution of %s completed in %,d ms", mc.jobIdString(), elapsed));
+            logger.info(formatExecutionSummary("completed successfully"));
             return true;
         }
         if (failure instanceof CancellationException || failure instanceof JobTerminateRequestedException) {
-            logger.info(String.format("Execution of %s completed in %,d ms, reason=%s",
-                    mc.jobIdString(), elapsed, failure));
+            logger.info(formatExecutionSummary(String.format("got terminated, reason=%s", failure)));
             return false;
         }
-        logger.severe(String.format("Execution of %s failed after %,d ms", mc.jobIdString(), elapsed), failure);
+        logger.severe(formatExecutionSummary("failed"), failure);
         return false;
     }
 
+    private String formatExecutionSummary(String conclusion) {
+        long executionEndTime = System.currentTimeMillis();
+        StringBuilder sb = new StringBuilder(String.format("Running %s %s", mc.jobIdString(), conclusion));
+        sb.append("\n\t").append("Start time: ").append(Util.toLocalDateTime(executionStartTime));
```

Comment:
is this start time or submission time

Reply:
It's set initially on `MasterJobContext` creation, which is the moment of submission, but it's also updated in `MasterJobContext.tryStartJob`, which makes it actual start time. When it gets printed on execution summary it's definitely START time. 

This is how it was before too, I just changed it from nano to millis time. I think we could remove the initialisation on submission.